src/main/java/millom/sandbox/service/MessageServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package millom.sandbox.service;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+public class MessageServiceImpl implements MessageService {
+
+  // format for ISO 8601
+  private static final DateTimeFormatter isoFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+  private static final LocalDate curiosityLandingDate = LocalDate.parse("2012-08-06",
+      isoFormat);
+
+  @Override
+  public String getHello() {
+    return "Hello World Jersey and HK2 ";
+  }
+
+  @Override
+  public int convertCuriositySol(String date) {
+
+    try {
+      LocalDate earthDate = LocalDate.parse(date, isoFormat);
```

Comment:
Maybe a check for `null` on date first? `parse()` will throw a `NullException` if `date` is set to `null`.

Reply:
If a check for `null` is added, then it is usually also a good idea to add a unit test for that specific case. But sometimes this can be a bit of an overkill. Good judgement from case to case is required here I think.

Reply:
Hi @kmmatwork , I am checking for "null" in convertCuriositySol() method in MyResource.java. So did not check here. Is that good idea to check in the both methods. or check only in any one? or which is better? please suggest me.